Framework/DataHandling/test/LoadSampleShapeTest.h

```diff
@@ -0,0 +1,211 @@
+  }
+
+  void testConfidence() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    std::string path = testLoadSampleShape.getPropertyValue("Filename");
+    auto *descriptor = new Kernel::FileDescriptor(path);
+    TS_ASSERT_EQUALS(90, testLoadSampleShape.confidence(*descriptor));
+    delete descriptor;
+  }
+
+  void testExec_2WS() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    prepareWorkspaces(testLoadSampleShape, "InputWS", "OutputWS");
+    TS_ASSERT_THROWS_NOTHING(testLoadSampleShape.execute());
+    TS_ASSERT(testLoadSampleShape.isExecuted());
+    clearWorkspaces("InputWS", "OutputWS");
+  }
+
+  void testExec_1WS() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    prepareWorkspaces(testLoadSampleShape, "InputWS", "InputWS");
+    TS_ASSERT_THROWS_NOTHING(testLoadSampleShape.execute());
+    TS_ASSERT(testLoadSampleShape.isExecuted());
+    clearWorkspaces("InputWS", "InputWS");
+  }
+
+  void test_output_workspace_has_MeshObject_2WS() {
+    loadMeshObject("InputWS", "OutputWS", "cube.stl");
+    clearWorkspaces("InputWS", "OutputWS");
+  }
+
+  void test_output_workspace_has_MeshObject_1WS() {
+    loadMeshObject("InputWS", "InputWS", "cube.stl");
+    clearWorkspaces("InputWS", "InputWS");
+  }
+
+  void test_cube() {
+    auto cube = loadMeshObject("InputWS", "InputWS", "cube.stl");
+    TS_ASSERT(cube->hasValidShape());
+    TS_ASSERT_EQUALS(cube->numberOfVertices(), 8);
+    TS_ASSERT_EQUALS(cube->numberOfTriangles(), 12);
+    TS_ASSERT_DELTA(cube->volume(), 3000, 0.001);
+    clearWorkspaces("InputWS", "InputWS");
+  }
+
+  void test_cylinder() {
+    auto cylinder = loadMeshObject("InputWS", "InputWS", "cylinder.stl");
+    TS_ASSERT(cylinder->hasValidShape());
+    TS_ASSERT_EQUALS(cylinder->numberOfVertices(), 722);
+    TS_ASSERT_EQUALS(cylinder->numberOfTriangles(), 1440);
+    TS_ASSERT_DELTA(cylinder->volume(), 589, 1);
+    clearWorkspaces("InputWS", "InputWS");
+  }
+
+  void test_tube() {
+    auto tube = loadMeshObject("InputWS", "InputWS", "tube.stl");
+    TS_ASSERT(tube->hasValidShape());
+    TS_ASSERT_EQUALS(tube->numberOfVertices(), 1080);
+    TS_ASSERT_EQUALS(tube->numberOfTriangles(), 2160);
+    TS_ASSERT_DELTA(tube->volume(), 7068, 1);
+    clearWorkspaces("InputWS", "InputWS");
+  }
+
+private:
+  // Create workspaces and add them to algorithm properties
+  void prepareWorkspaces(LoadSampleShape &alg, const std::string &inputWS,
+                         const std::string &outputWS) {
+    auto inWs =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(10, 4);
+    Mantid::API::AnalysisDataService::Instance().add(inputWS, inWs);
```

Comment:
The use of singletons (here the `ADS`), can and should be avoided here. You do not need to put the workspace in the ADS and do not need to clear it out if you set the algoritm...

```cpp
alg.setChild(true);
```
Then you can simply set and fetch the Workspace directly, i.e.
```cpp
alg.setProperty("InputWorkspace", inputWS);
...
MatrixWorkspace_sptr outWS = alg.getProperty("OutputWorkspace");
```
Please refactor the use of the ADS out of this test suite.

Reply:
I'm great difficulty in getting this to work without ADS, which does the useful task of setting up an input workspace. If I try this using a workspace pointer of a workspace created by a helper function, it crashes when I attempt to `setProperty` of the algorithm to it. Do you know of any similar algorithm test that avoids the ADS as you suggest? I'll spend some time looking for one myself.

Reply:
Karl, I'm away next week, perhaps @martyngigg or another more experienced developer in the office could help you with this.